tests/foreman/api/test_bookmarks.py

```diff
@@ -0,0 +1,290 @@
+        (not listed)
+
+        @Status: Manual
+        """
+        for entity in BOOKMARK_ENTITIES:
+            with self.subTest(entity['controller']):
+                pass
+
+    @stubbed()
+    @tier1
+    def test_negative_create_bookmark_no_querye(self):
+        """Create a bookmark with empty query
+
+        @Feature: Scoped Search Bookmark Create
+
+        @Steps:
+
+        1. Create a bookmark with a providing no query
+        2. List the bookmarks
+
+        @Assert: Error notification - search query cannot be empty, Bookmark is
+        not created (not listed)
+
+        @Status: Manual
+        """
+        for entity in BOOKMARK_ENTITIES:
+            with self.subTest(entity['controller']):
+                pass
+
+    @stubbed()
+    @tier1
+    def test_negative_create_bookmark_same_name(self):
+        """Create bookmarks with the same names
+
+        @Feature: Scoped Search Bookmark Create
+
+        @Setup:
+
+        1. Create a bookmark of a random name
+
+        @Steps:
+
+        1. Create a bookmark with providing the same name as the pre-created
+        one
+        2. List the bookmarks
+
+        @Assert: Error notification - name already taken, Bookmark is not
+        created (not listed)
+
+        @Status: Manual
+        """
+        for entity in BOOKMARK_ENTITIES:
+            with self.subTest(entity['controller']):
+                pass
+
+    @stubbed()
+    @tier1
+    def test_negative_create_bookmark_public_null(self):
+        """Create a bookmark with omitting the public parameter
+
+        @Feature: Scoped Search Bookmark Create
+
+        @Steps:
+
+        1. Create a bookmark of a random name, random query and omit the
+        'public' parameter
+        2. List the bookmarks
+
+        @Assert: Error notification – public is required, Bookmark is not
+        created (not listed)
+
+        @Status: Manual
+        """
+        for entity in BOOKMARK_ENTITIES:
+            with self.subTest(entity['controller']):
+                pass
+
+    # UPDATE TESTS
+    @stubbed()
+    @tier1
+    def test_positive_update_bookmark_name(self):
+        """Update a bookmark
+
+        @Feature: Scoped Search Bookmark Update
+
+        @Setup:
+
+        1. Create a bookmark of a random name with random query
+
+        @Steps:
+
+        1. Update the name of the pre-created bookmark
+
+        @Assert: The new bookmark name is listed
+
+        @Status: Manual
+        """
+        for entity in BOOKMARK_ENTITIES:
+            with self.subTest(entity['controller']):
+                pass
+
+    @stubbed()
+    @tier1
+    def test_negative_update_bookmark_name(self):
+        """Update a bookmark with name already taken
+
+        @Feature: Scoped Search Bookmark Update
+
+        @Setup:
+
+        1. Create 2 bookmarks of random names with random query
+
+        @Steps:
+
+        1. Update the name to one of the other pre-created bookmarks
```

Comment:
1. Try to update the name of the first (or second) Bookmark created in the Setup with the name of the second (or first) Bookmark